model.py

```diff
@@ -25,66 +25,57 @@ class Genre(db.Model):
 
     __tablename__ = "genres"
 
-    genre_id = db.Column(db.Integer,
-                        autoincrement=True,
-                        primary_key=True)
+    genre_id = db.Column(db.Integer, primary_key=True)
     name = db.Column(db.String(64), nullable=True)
 
     def __repr__(self):
         """Provide helpful representation when printed."""
 
         return f"<Genre genre_id={self.genre_id} name={self.name}>"
 
-# association_table = db.Table('association', 
-#     db.Column('location_id', db.Integer, db.ForeignKey('locations.location_id'), primary_key=True),
-#     db.Column('movie_id', db.Integer, db.ForeignKey('movies.movie_id'),primary_key=True),  
-# )
-#maybe explore this more later
+
+association_table = db.Table('movies_locations',
+                             db.Column('location_id', db.Integer, db.ForeignKey('locations.location_id')),
+                             db.Column('movie_id', db.Integer, db.ForeignKey('movies.movie_id')),
+                             )
+
 
 class Movie(db.Model):
     """Movie on ratings website."""
 
     __tablename__ = "movies"
 
-    movie_id = db.Column(db.Integer,
-                         autoincrement=True,
-                         primary_key=True)
+    movie_id = db.Column(db.Integer, primary_key=True)
     title = db.Column(db.String(100))
     genre_id = db.Column(db.Integer, db.ForeignKey('genres.genre_id'), nullable=False)
-
-    
     genre = db.relationship('Genre', backref='movies')
-    location_ids = db.Column(ARRAY(db.Integer))
-    # locations = db.relationship('Location', 
-    #     secondary=association_table, lazy='subquery',
-    #     backref=db.backref('movies', lazy=True))
-
 
+    # To add locations: movie.locations.append(location); then save as usual using db.session.add/commit
+    locations = db.relationship('Location',
+                                secondary='movies_locations')
 
     def __repr__(self):
         """Provide helpful representation when printed."""
 
-        return f"<Movie movie_id={self.movie_id} title={self.title} genre={self.genre} locations={self.location_ids}"
+        return f"<Movie movie_id={self.movie_id} title={self.title} genre={self.genre} locations={self.locations}"
 
 
 class Location(db.Model):
     """Location of a user."""
 
     __tablename__ = "locations"
 
-    location_id = db.Column(db.Integer,
-                          autoincrement=True,
-                          primary_key=True)
+    location_id = db.Column(db.Integer, primary_key=True)
     latitude = db.Column(db.Float)
     longetude = db.Column(db.Float)
     name = db.Column(db.String())
+    # TODO: Make sure this is a full url with http:// appended, otherwise must add this in templates manually!
     href = db.Column(db.String(2000))
 
-
     def __repr__(self):
         """Provide helpful representation when printed."""
 
-        return f"<Location location_id={self.location_id} latitude={self.latitude} longitude={self.longitude} name={self.name}>"
```

Comment:
Typos that caused an error in your app. You may want to fix it to the proper spelling 'longitude' just to avoid future bugs.